src/commands.rs

```diff
@@ -4009,11 +4030,18 @@ fn is_fast_forward(repo: RepoRef, branch_name: &str, new_target_id: &CommitId) -
     }
 }
 
-fn cmd_branch(ui: &mut Ui, command: &CommandHelper, args: &BranchArgs) -> Result<(), CommandError> {
+fn cmd_branch(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &BranchSubcommand,
+) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let branch_names: Vec<&str> = if args.delete || args.forget {
-        let view = workspace_command.repo().view();
-        args.names
+    let view = workspace_command.repo().view();
+    fn assert_branch_names_exist<'a>(
```

Comment:
It looks like this either returns all the input names in the input or an error. If I understood that correctly, then it seems better to have it return `Result<(), CommandError>` so it's clear that no filtering is going on.